Tools/autotest/autotest.py

```diff
@@ -651,8 +651,6 @@ def list_subtests(*args, **kwargs):
     ''' main program '''
     os.environ['PYTHONUNBUFFERED'] = '1'
 
-    os.putenv('TMPDIR', util.reltopdir('tmp'))
```

Comment:
Why does this break MacOSX stuff?

This wouldn't have been put here for fun, I'm sure someone needed this.  For example, on the autotest server we use a specific partition for building for speed, using the default `TMPDIR` may be slow.

Reply:
@peterbarker the ar on mac doesn't like setting of TMPDIR. mac os has its own system of tmpdir happening. I can put this back in for other systems. I removed it because you said it wasn't necessary.

Reply:
I have put this back in for systems other than macos